### DISCUSSION TITLE
2.3.13

### DISCUSSION THREAD

**Discussion by benhillis** [microsoft/WSL#11839] [General]
*  Docker desktop regression introduced in 2.3.11 fixed.
*  Properly initialize file offsets in the relaying logic (solves #11799)

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/WSL/releases/tag/2.3.13'>2.3.13</a>.</em>

**Comment by astroboylrx** [comment]
Microsoft Store forced an update on WSL despite "App updates" turned off in the settings. The update was so abrupt and killed everything... I was still using the distro.

Can we users humbly request an option to choose a time for update? Or how may I prevent such events?

**Comment by FlattusBlastus** [comment]
Had to revert to 2.2.4.0. Docker Desktop for Windows will not launch.

**Reply by FlattusBlastus** [reply]
Reverting didnt fix it. I had to uninstall Docker Desktop for Windows entirely then reinstall it with the latest: 4.33.0. I now see this:
C:\Windows\System32>wsl --version
WSL version: 2.3.13.0
Kernel version: 6.6.36.3-1
WSLg version: 1.0.63
MSRDC version: 1.2.5326
Direct3D version: 1.611.1-81528511
DXCore version: 10.0.26100.1-240331-1435.ge-release
Windows version: 10.0.26100.1297

Docker for Desktop loads correctly now.

**Comment by stephanwehr** [comment]
Also had to downgrade from version 2.3.13 to 2.2.4.
After a reboot I was able to start Docker Desktop 4.33.0, but Kubernetes container couldn't communicate with each other.
I cannot confirm if DNS or network is affected.

Windows 11 X86

**Comment by suprsidr** [comment]
> * Docker desktop regression introduced in 2.3.11 fixed.
> * Properly initialize file offsets in the relaying logic (solves [appending to a file via shell redirecting (echo foo >> bar) replaces content from start of document, doesn't append #11799](https://github.com/microsoft/WSL/issues/11799))
> 
> _This discussion was created from the release [2.3.13](https://github.com/microsoft/WSL/releases/tag/2.3.13)._

Still having this issue:
<img width="545" alt="wsl-docker-error" src="https://github.com/user-attachments/assets/d164e4c1-5a38-4abb-8cc9-2539607b64de">
Windows 11 ARM

docker dump id: `6C8183BA-69B8-4A2F-BA95-CAEE6241F043/20240802202936`

BTW this hoses my whole setup and I cannot recover.
2.3.11 forced me to reset my whole PC and reinstall everything. 
I tried rollback and every other suggestion to no avail.

Today I'm about to do the same :(
